### PR TITLE
IDAM-466 Delete Organisation + IDAM-462 View Organisation Details Page

### PR DESCRIPTION
**Pull request by suegarner**
https://nicedigital.atlassian.net/browse/IDAM-466

### CODE REVIEW

File: NICE.Identity.Management/ClientApp/src/components/App/App.tsx [hunk 1]
```diff
@@ -54,6 +58,19 @@ export class App extends React.Component {
...
+					<Route path="/organisations/:id" exact component={Organisation} />
+					{/* <Route
+						path={"/organisations/:id"}
+						render={({ location }) => {
+							// to stop rendering of component for 'add' route
+							const idRegExp = new RegExp(/[0-9]+$/g);
+							const endOfRoute = location.pathname.split("/").pop() ?? "";
+							return idRegExp.test(endOfRoute) && <Organisation />;
+						}}
+					/> */}
```
Comment (johnholland-nice) [inline NICE.Identity.Management/ClientApp/src/components/App/App.tsx:72]: I found the proper solution to the issue that we were experiencing with the commented out route stuff. Could we replace with the following:

```suggestion
					<Route
						path={"/organisations/:id"}
						render={(props) => {
							// to stop rendering of component for 'add' route
							const idRegExp = new RegExp(/[0-9]+$/g);
							const endOfRoute = props.location.pathname.split("/").pop() ?? "";
							return idRegExp.test(endOfRoute) && <Organisation {...props} />;
						}}
					/>
```

File: NICE.Identity.Management/ClientApp/src/views/DeleteOrganisation/DeleteOrganisation.tsx [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+		this.setState({ isLoading: true });
+
+		const organisation = await fetchData(Endpoints.organisation(this.props.match.params.id));
+
+		if (isDataError(organisation)) {
+			this.setState({ error: organisation });
+		}
+
+		this.setState({ organisation, isLoading: false });
+		document.title = `NICE Accounts - Delete ${organisation.name}`;
```
Comment (johnholland-nice) [inline NICE.Identity.Management/ClientApp/src/views/DeleteOrganisation/DeleteOrganisation.tsx:74]: This should go in the bottom of the constructor above.

File: NICE.Identity.Management/ClientApp/src/views/Organisation/Organisation.tsx [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+
+		async componentDidMount(): Promise<void> {
+			this.setState({ isLoading: true });
+	
+			const organisation = await fetchData(Endpoints.organisation(this.props.match.params.id));
+			const users = await fetchData(Endpoints.usersByOrganisation(this.props.match.params.id));
+
+			if (isDataError(users)) {
+				this.setState({ error: users });
+			}
```
Comment (johnholland-nice) [inline NICE.Identity.Management/ClientApp/src/views/Organisation/Organisation.tsx:60]: I think we need to check organisation against isDataError too here like we do for users.

File: NICE.Identity.Management/ClientApp/src/views/Organisation/Organisation.tsx [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+								</Button>
+								<Button
+									data-qa-sel="edit-organisations-users-button"
+									to={`/users-organisations/${this.props.match.params.id}/edit`}
+									elementType={Link}
+									disabled={isLoading}
+								>
+									Edit users
+								</Button>
+							</>
```
Comment (johnholland-nice) [inline NICE.Identity.Management/ClientApp/src/views/Organisation/Organisation.tsx:100]: Do we maybe need to hide these until the pages exist? Not an issue for me, but wondering if that might come back from the testers...

File: NICE.Identity.Management/ClientApp/src/views/Organisation/Organisation.tsx [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+import { Alert } from "@nice-digital/nds-alert";
+
+import { fetchData } from "../../helpers/fetchData";
+import { isDataError } from "../../helpers/isDataError";
+import { Endpoints } from "../../data/endpoints";
+
+import { Button } from "@nice-digital/nds-button";
+
+import { ErrorMessage } from "../../components/ErrorMessage/ErrorMessage";
+import styles from "./Organisation.module.scss";
```
Comment (johnholland-nice) [inline NICE.Identity.Management/ClientApp/src/views/Organisation/Organisation.tsx:18]: Did we add this file? Can't see it, so sorry if so.

File: NICE.Identity.Management/ClientApp/src/views/Organisation/Organisation.tsx [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+								</Button>
+							</>
+					<Grid>
+						<GridItem cols={12} md={9} aria-busy={isLoading}>
+									<div className={`${styles.summaryList} pv--c`}>
+										<span className={styles.summaryListLabel}>
+											Date added
+										</span>
+										<span
+											className={styles.summaryListDetail}
```
Comment (johnholland-nice) [inline NICE.Identity.Management/ClientApp/src/views/Organisation/Organisation.tsx:108]: Not sure if styles exists if the Organisation.module.scss doesn't exist - see comment near the top.

File: NICE.Identity.Management/ClientApp/src/views/Organisation/Organisation.tsx [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+									elementType={Link}
+									disabled={isLoading}
+								>
+									Edit users
+								</Button>
+							</>
+					<Grid>
+						<GridItem cols={12} md={9} aria-busy={isLoading}>
+									<div className={`${styles.summaryList} pv--c`}>
+										<span className={styles.summaryListLabel}>
```
Comment (johnholland-nice) [inline NICE.Identity.Management/ClientApp/src/views/Organisation/Organisation.tsx:104]: Not sure if styles exists if the Organisation.module.scss doesn't exist - see comment near the top.

File: NICE.Identity.Management/ClientApp/src/views/Organisation/Organisation.tsx [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+											{ToFormattedDateString(organisation.dateAdded)}
+										</span>
+									</div>
+						</GridItem>
+
+						<GridItem cols={12} md={9} aria-busy={isLoading}>
+						{isLoading ? (
+								<p>Loading...</p>
+							) : users.length ? (
+									<div className={`${styles.summaryList} pv--c`}>
```
Comment (johnholland-nice) [inline NICE.Identity.Management/ClientApp/src/views/Organisation/Organisation.tsx:120]: Not sure if styles exists if the Organisation.module.scss doesn't exist - see comment near the top.

File: NICE.Identity.Management/ClientApp/src/views/Organisation/Organisation.tsx [hunk 9]
```diff
@@ -0,0 +1,166 @@
...
+										<span className={styles.summaryListLabel}>
+											Users
+										</span>
+										<span className={styles.summaryListDetail}>
```
Comment (johnholland-nice) [inline NICE.Identity.Management/ClientApp/src/views/Organisation/Organisation.tsx:124]: I'll stop copying the styles comment, but basically if we don't have a Organisation.module.scss file, we can remove all the references to the styles.summary class stuff or alternatively we need to create a module file that has the same copied styles from Users.module.scss.

File: NICE.Identity.Management/ClientApp/src/views/Organisation/Organisation.tsx [hunk 10]
```diff
@@ -0,0 +1,166 @@
...
+											<ul className="list--unstyled" data-qa-sel="list-of-users">
```
Comment (johnholland-nice) [inline NICE.Identity.Management/ClientApp/src/views/Organisation/Organisation.tsx:125]: I think the data-qa-sel value here needs to be unique. I think it's been left in from another component.

File: NICE.Identity.Management/ClientApp/src/views/Organisation/Organisation.tsx [hunk 11]
```diff
@@ -0,0 +1,166 @@
...
+													<li key={user.userId}>
+														<UserStatus user={user} />
+														<span className={styles.userName}>{user.firstName + " " + user.lastName}</span>
+													</li>
+												);
+											})}
+											</ul>
+										</span>
+									</div>
+							) : <p>Three are currently no users assigned to this oragnisation</p>
```
Comment (johnholland-nice) [inline NICE.Identity.Management/ClientApp/src/views/Organisation/Organisation.tsx:137]: Spelling mistake

File: NICE.Identity.Management/ClientApp/src/views/Organisation/Organisation.tsx [hunk 12]
```diff
@@ -0,0 +1,166 @@
...
+		constructor(props: OrganisationProps) {
+			super(props);
+	
+			this.state = {
+				organisation: {} as OrganisationType,
+				users: [],
+				redirect: false,
+				isLoading: true,
+			};
+		}
```
Comment (johnholland-nice) [inline NICE.Identity.Management/ClientApp/src/views/Organisation/Organisation.tsx:50]: Could we add a document title in? Think this is missing off other components, so apologies if it was missing in the one you used as reference.

```suggestion
		constructor(props: OrganisationProps) {
			super(props);
	
			this.state = {
				organisation: {} as OrganisationType,
				users: [],
				redirect: false,
				isLoading: true,
			};
			
			document.title = "NICE Accounts - Organisation details";
		}
```

**Review by johnholland-nice — CHANGES_REQUESTED** [review]
Left a few comments; should be straightforward, but give me a shout if you need to discuss. Looks good though! :)

File: NICE.Identity.Management/ClientApp/src/views/DeleteOrganisation/DeleteOrganisation.tsx (hunk 2 above)
Comment (suegarner) [reply NICE.Identity.Management/ClientApp/src/views/DeleteOrganisation/DeleteOrganisation.tsx:74]: Leaving it in component did mount as constructor doesn't have org name yet

File: NICE.Identity.Management/ClientApp/src/views/Organisation/Organisation.tsx [hunk 13]
```diff
@@ -0,0 +1,166 @@
...
+							) : <p>Three are currently no users assigned to this oragnisation</p>
+						}
+						</GridItem>
+					</Grid>
+					<hr className="mv--b" />
+
+					<h2 className="h3">Permanently delete this organisation</h2>
+					<p>
+						This organisation will no longer be available, and all data in associated with the
+						organisation will be permanently deleted.
```
[the remark this answers is not in the document]
Comment (imranazad) [reply NICE.Identity.Management/ClientApp/src/views/Organisation/Organisation.tsx:146]: @suegarner This one is for you.

**Review by johnholland-nice — DISMISSED** [review]

**Review by johnholland-nice — DISMISSED** [review]

**Review by johnholland-nice — APPROVED** [review]

**Review by johnholland-nice — APPROVED** [review]

**Pull request merged**